mmdet/models/losses/kd_loss.py

```diff
@@ -0,0 +1,93 @@
+import mmcv
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..builder import LOSSES
+from .utils import weighted_loss
+
+
+@mmcv.jit(derivate=True, coderize=True)
+@weighted_loss
+def knowledge_distillation_loss(pred, soft_label, T, detach_target=True):
+    r"""Loss function of `Distilling the Knowledge in a Neural Network.
+    <https://arxiv.org/abs/1503.02531>`_.
+
+    Args:
+        pred (torch.Tensor): Predicted general distribution of bounding boxes
+            (before softmax) with shape (N, n+1), n is the max value of the
+            integral set `{0, ..., n}` in paper.
+        soft_label (torch.Tensor): Target soft label learned from teacher
+            with shape (N, n+1).
+        T (int): Temperature for distillation.
+
+    Returns:
+        torch.Tensor: Loss tensor with shape (N,).
+    """
+
+    target = F.softmax(soft_label / T, dim=1)
+    if detach_target:
+        target = target.detach()
+
+    kd_loss = F.kl_div(
+        F.log_softmax(pred / T, dim=1), target, reduction='none').mean(1) * (
+            T * T)
+
+    return kd_loss
+
+
+@LOSSES.register_module()
+class LocalizationDistillationLoss(nn.Module):
```

Comment:
Better rename it as `KnowledgeDistillationKLDivLoss` as it is general and can also be applied to classification tasks as well.